### PR TITLE
Add UI and API for the new shallow clone option in git materials

### PR DESCRIPTION
**Pull request by wpc**
for issue #1979

### CODE REVIEW

File: server/webapp/localize.xml [hunk 1]
```diff
@@ -377,6 +377,7 @@
     <entry key="BLACKLIST">Blacklist</entry>
     <entry key="BLACKLIST_DESCRIPTION">Enter the paths to be excluded. Separate multiple entries with a comma.</entry>
     <entry key="BRANCH">Branch</entry>
+    <entry key="SHALLOW_CLONE">Shallow clone (recommended for large repository)</entry>
```
Comment (ketan) [inline server/webapp/localize.xml]: `recommended for a large repository` or `recommended for large repositories`

**Comment by wpc** [comment]
This PR is ready for review.

File: server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/materials/git_material_representer.rb [hunk 2]
```diff
@@ -27,6 +27,7 @@ class GitMaterialRepresenter < ScmMaterialRepresenter
                    value.blank? ? self.setBranch('master') : self.setBranch(value)
                  }
         property :submodule_folder
+        property :shallow_clone
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/materials/git_material_representer.rb]: You will also need to submit a PR to https://github.com/gocd/api.go.cd to document this property, and clearly indicate that it was introduced in 16.3.0. See `agent_state` or `build_state` attribute at https://api.go.cd/current/#the-agent-object for an example.

Comment (wpc) [inline server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/materials/git_material_representer.rb]: The document PR: https://github.com/gocd/api.go.cd/pull/73

Thx for pointing this out.

**Comment by ketan** [comment]
:+1: 

Please squash and let me know.

**Comment by jyotisingh** [comment]
Looks good to me

**Comment by wpc** [comment]
@ketan, @jyotisingh  I have squashed the changes.

**Comment by jyotisingh** [comment]
@wpc - one more thing. User docs needs to be updated too => https://www.go.cd/documentation/user/current/configuration/configuration_reference.html#git

Can we also point out that `shallowClone` applies to agent-side alone. Checkout on Server side ie. flyweight  is always shallow-cloned or something along those lines.

**Comment by wpc** [comment]
hi @jyotisingh: 
for that do I send PR to https://github.com/gocd/docs.go.cd or https://github.com/gocd/documentation?

**Comment by ketan** [comment]
@juhijariwala is splitting gocd/documentation into docs.go.cd and developer.go.cd. Submitting an issue against docs.go.cd and submit a PR later is probably best to avoid weird conflicts

**Pull request merged**